Backend/app/provider-platform/static-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -251,60 +251,69 @@ rideRoute merchantShortId reqRideId = do
 ---------------------------------------------------------------------
 
 ---------------------------------------------------------------------
-rideSync :: ShortId DM.Merchant -> Id Common.Ride -> Flow Common.RideSyncRes
-rideSync merchantShortId reqRideId = do
+rideSync :: ShortId DM.Merchant -> Common.RideSyncReq -> Flow Common.RideSyncRes
+rideSync merchantShortId rideSyncReq = do
   merchant <- findMerchantByShortId merchantShortId
-  let rideId = cast @Common.Ride @DRide.Ride reqRideId
-  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
-  booking <- runInReplica $ QBooking.findById ride.bookingId >>= fromMaybeM (BookingNotFound rideId.getId)
+  let rideIds = map (cast @Common.Ride @DRide.Ride) rideSyncReq.rideIds
+  rides <- mapM (\rideId -> runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)) rideIds
```

Comment:
Why should we fetch each ride from DB separately? I think it increase DB load. Can you fetch all rides at once (see `stuckBookingsCancel` endpoint for example)?

Reply:
fixed it